.clinerules/msal-cline-rules.md

```diff
@@ -0,0 +1,167 @@
+- If user does not specify between multiple account mode and single account mode in MSAL, the default mode used should be MULTIPLE. Make sure the configuration json includes which account mode the application is being built for, and use the appropriate example based on this selection.
+- Ensure that the latest MSAL Version is used. Reference the MSAL repo to see latest version. At the very least, MSAL 6.+ should be used.
+- Cline must copy the following gradle files exactly when creating a new application:
+  1. Root level build.gradle (project level) from examples/hello-msal-multiple-account/build.gradle or examples/hello-msal-single-account/build.gradle
+  2. App level build.gradle from examples/hello-msal-multiple-account/app/build.gradle or examples/hello-msal-single-account/app/build.gradle
+  3. gradle.properties from examples/hello-msal-multiple-account/gradle.properties or examples/hello-msal-single-account/gradle.properties
+  4. settings.gradle from examples/hello-msal-multiple-account/settings.gradle or examples/hello-msal-single-account/settings.gradle
+
+  These files must be placed in their corresponding locations in the new application's directory structure. The only modification allowed is updating the applicationId and namespace in app/build.gradle and rootProject.name in settings.gradle to match the new application's package name. Cline can divert from these files as needed once initial creation is complete.
+
+# MSAL API Usage
+- Broker integration (Microsoft Authenticator, Company Portal, or Link To Windows) should ALWAYS be enabled for enhanced security and SSO capabilities. The broker_redirect_uri_registered field in auth_config.json should be set to "true" unless explicitly disabled by the user's request.
+- ALWAYS use MSAL's Parameters-based APIs instead of deprecated methods. Here are the required patterns:
+
+For interactive token acquisition in multiple account applications:
+```java
+AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+    .withScopes(SCOPES)
+    .withCallback(callback)
+    .build();
+mPCA.acquireToken(parameters);
+```
+
+For silent token acquisition:
+```java
+AcquireTokenSilentParameters parameters = new AcquireTokenSilentParameters.Builder()
+    .withScopes(SCOPES)
+    .forAccount(account)
+    .forceRefresh(false)
+    .withCallback(callback)
+    .build();
+mPCA.acquireTokenSilent(parameters);
+```
+
+For sign-in in single account applications only:
+```java
+SignInParameters parameters = new SignInParameters.Builder()
+    .startActivity(activity)
+    .withCallback(callback)
+    .build();
+mPCA.signIn(parameters);
+```
+
+Important notes:
+- For multiple account applications, use acquireToken for sign in
+- For single account applications, use signIn
```

Comment:
Once signed In, subsequent requests will use acquireToken even for Single Account, right?

Reply:
iirc signInAgain is meant to be used in place of acquireToken for single account.

Reply:
signInAgain snippet added